plugins/parsers/carbon2/parser.go

```diff
@@ -46,7 +46,8 @@ func (p Parser) Parse(buf []byte) ([]telegraf.Metric, error) {
 
 		m, err := parseLine(line)
 		if err != nil {
-			return nil, fmt.Errorf("failed to parse line: %s, err: %w", line, err)
+			fmt.Printf("E! Failed parsing Carbon2 line with metric: %s\n", line)
+			continue
```

Comment:
could we store all errors in one and return it at the end?

Reply:
We could but then none of the metrics are send out due to the error

```
2021-07-29T12:43:58Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
2021-07-29T12:44:00Z E! [inputs.file] Error in plugin: 1 error occurred:
        * failed to parse value as int: NaN , err: strconv.ParseInt: parsing "NaN": invalid syntax

2021-07-29T12:44:00Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
```

vs just printing out and leaving the error as is.

```
2021-07-29T12:40:40Z D! [agent] Starting service inputs
E! Failed parsing Carbon2 line with metric: className=HealthTrackerXXX cluster=XXX deployment=XXX ... stat=p75  _primaryMetricType=carbon NaN 1625855958
2021-07-29T12:40:43Z D! [outputs.file] Wrote batch of 2 metrics in 362.274µs
```